### ISSUE TITLE
Including GIT commit hash causes full re-build every time I commit

### ISSUE THREAD

**Issue by tjayrush** [Great-Hill-Corporation/quickBlocks#1129]
In our CMakeLists we generate some #defines for the GIT_COMMIT and GIT_TIMESTAMP. This is excellent (I love it), but it forces the entire c++ tree to re-build every time I make a commit. (Every file depends on the #define (-D)).

If instead, the CMakeList created a .cpp file with these values defined (through some sort of shell script for example) then only that file would re-build when the commit changes. The file (called `./utillib/git_commit.cpp`) could look something like this:

```
#include "version.h"
namespace qblocks {
    string_q GIT_COMMIT "230202093029"
    timestamp_t GIT_TIMESTAMP 122321433242
}
```

and then in the versions.h header file thus:

```
namespace qblocks {
    extern string_q GIT_COMMIT;
    extern timestamp_t GIT_TIMESTAMP
}
```

Any .cpp file could use the values, but only the `git_version.cpp` would rebuild on a new commit.

**Comment by tjayrush** [comment]
This may be informative, although this discusses the makefile not the CMakeLists file. https://stackoverflow.com/questions/51727566/how-to-make-git-commit-hash-available-in-c-code-without-needless-recompiling

**Comment by tjayrush** [comment]
This is fixed with commit a9fb44ca...
